lib/timesheet.rb

```diff
@@ -47,7 +55,7 @@ def valid_attribute?
     end
 
     def client_valid?
-      unless client_field_still_na? || client_field_still_invalid?
+      unless client_field_still_na? || client_field_still_invalid? || project_not_billable?
```

Comment:
This check is a mess. Two big things:

* Don't use `unless`. It's clearer to say "if not" than "unless"
* Don't `return true` or `false`. The expression will evaluate true or false, and if that's the last line of the method, that value will get returned.